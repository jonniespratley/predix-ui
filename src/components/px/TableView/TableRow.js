import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import Icon from '../IconSet/Icon';

const TableRow = styled.div`
  user-select: none;
  min-height: var(--px-table-row-min-height, 2.93333rem);
  line-height: var(--px-table-row-line-height, 15px);
  color: var(--px-table-row-color, inherit);
  font-size: var(--px-table-row-font-size, 15px);
  background-color: var(--px-table-row-background-color, white);
  border-bottom-width: var(--px-table-row-separator-width, 1px);
  border-bottom-style: var(--px-table-row-separator-style, solid);
  border-bottom-color: var(--px-table-row-separator-color, black);
  user-select: none;
  align-items: center;
  box-sizing: border-box;
  position: relative;
  overflow: hidden;
  display: flex;
  flex-direction: row;
  appearance: normal;
  flex: 1;

  &:after,
  &:before{
    display: table;
    content: ' ';
  }

  ${props => props.tappable && css`
    cursor: pointer;
    &:active {
      color: var(--px-table-row-selected-color, white) !important;
      background: var(--px-table-row-selected-background-color, black);
    }
  `}

  ${props => props.selected && css`
    background-color: var(--px-table-row-selected-background-color, black) !important;
    color: var(--px-table-row-selected-color, white) !important;
    .px-table-row-body{
      color: inherit;
    }
  `}

  ${props => props.header && css`
    padding: var(--px-table-row-header-padding, 0.33333rem);
    font-size: var(--px-table-row-header-font-size, 15px);
    color: var(--px-table-row-header-color, inherit);
    background-color: var(--px-table-row-header-background-color, black);
    text-transform: var(--px-table-row-header-text-transform, uppercase);
    min-height: var(--px-table-row-header-min-height, 24px);
    line-height: var(--px-table-row-header-line-height, 24px);
    border-top: none;
    border-bottom: none;
  `}

  ${props => props.swipeable && css`
    background-color: var(--px-table-row-background-color, white);
    overflow: hidden;
    transform-style: preserve-3d;
    transition: transform cubic-bezier(0.55, 0, 0.1, 1) 0.3s;
    z-index: 10;
  `}

  ${props => props.size === 'flush' && css`
    padding: 0;
  `}
  ${props => props.size === 'tiny' && css`
    padding: 0.33333rem;
  `}
  ${props => props.size === 'small' && css`
    padding: 0.66667rem;
  `}
  ${props => props.size === 'regular' && css`
    padding: 1rem;
  `}
  ${props => props.size === 'large' && css`
    padding: 1.33333rem;
  `}
  ${props => props.size === 'huge' && css`
    padding: 2rem;
  `}
`;
TableRow.displayName = 'TableRow';

const TableRowTitle = styled.div`
  user-select: none;
  line-height: normal;
  flex: 1 0 auto;
  align-items: center;
  align-self: stretch;
  display: flex;
  white-space: nowrap;
  font-size: var(--px-table-row-title-font-size, 15px);
  color: var(--px-table-row-title-color, inherit);
`;
TableRowTitle.displayName = 'TableRowTitle';

const TableRowBody = styled.div`
  font-size: var(--px-table-row-body-font-size, 15px);
  order: 5;
  align-self: stretch;
  flex: 1 0 auto;
  white-space: normal;
  line-height: normal;
`;
TableRowBody.displayName = 'TableRowBody';

const TableRowMedia = styled.div`
  flex: 0 1 auto;
  order: 0;
  display: flex;
  align-self: stretch;
  justify-content: center;
  align-content: center;
  flex-direction: column;
  margin-left: var(--px-table-row-media-margin, 0);
  margin-right: var(--px-table-row-media-margin, 0.66667rem);

  img {
    display: block;
    max-width: 80px;
    max-height: 80px;
  }
`;
TableRowMedia.displayName = 'TableRowMedia';

const TableRowLabel = styled.div`
  -webkit-touch-callout: none;
  user-select: none;
  order: 0;
  align-self: center;
  white-space: nowrap;
  position: relative;
  top: 2px;

  ${props => props.labelLeft && css`
    flex: 0 1 0;
    text-align: right;

  `}

  ${props => props.labelRight && css`
    text-align: left;
    order: 4;

  `}
`;
TableRowLabel.displayName = 'TableRowLabel';

const TableRowSubtitle = styled.div`
  font-size: var(--px-table-row-subtitle-font-size, 15px);
  color: var(--px-table-row-subtitle-color, black);
`;

const TableRowContent = styled.div`
  user-select: none;
  display: flex;
  flex-direction: column;
  align-self: stretch;

  flex: 1;
  user-select: none;
`;
TableRowContent.displayName = 'TableRowContent';

const TableRowActions = styled.div`
  overflow: hidden;
  transform-style: preserve-3d;
  position: absolute;
  top: 0;
  bottom: 0;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: auto;
  align-self: stretch;
  align-items: stretch;

  ${props => props.opened && css`
    z-index: 50;
  `}
  ${props => props.actionsLeft && css`
    order: 0;
    left: 0;
    ${props.opened && css`
      width: auto;
      transform: translateX(0%);
    `}
  `}
`;

/**
 * px-table-row component
 * https://github.com/PredixDev/px-table-view/blob/master/px-table-row.html
 */
const TableRowComponent = ({
  title,
  subtitle,
  body,
  image,
  icon,
  selected,
  labelLeft,
  labelRight,
  editMode,
  size,
  tappable,
  swipeable,
  header,
  underlayContent,
  rowContent,
  actions,
  children
}) => (
  <TableRow
    className="px-table-row"
    size={size}
    tappable={tappable}
    selected={selected}
    swipeable={swipeable}
    header={header}
  >
    {labelLeft && <TableRowLabel labelLeft className="table-row__label table-row__label--left">{labelLeft}</TableRowLabel>}
    {image && <TableRowMedia><img src={image} alt={title} /></TableRowMedia>}
    {icon && <TableRowMedia><Icon icon={icon} /></TableRowMedia>}
    {editMode && <div className="table-row__media table-row__media--icon table-row__media--right"><button className="btn btn--bare table-row__handle">hamburger</button></div>}
    {title &&
    <TableRowContent>
      {title && <TableRowTitle >{title}</TableRowTitle>}
      {subtitle && <TableRowSubtitle>{subtitle}</TableRowSubtitle>}
      {body && <TableRowBody>{body}</TableRowBody>}
      {rowContent}
    </TableRowContent>}

    {labelRight && <TableRowLabel labelRight className="table-row__label table-row__label--right">{labelRight}</TableRowLabel>}
    {children && <TableRowContent>{children}</TableRowContent>}
    {actions && <TableRowActions>{actions}</TableRowActions>}
    <div id="underlay" className="flex flex--stretch">
      {underlayContent}
    </div>
  </TableRow>
);

TableRowComponent.propTypes = {
  tappable: PropTypes.bool,
  swipeable: PropTypes.bool,
  selected: PropTypes.bool,
  header: PropTypes.bool,
  editMode: PropTypes.bool,
  size: PropTypes.string,
  title: PropTypes.string,
  subtitle: PropTypes.string,
  icon: PropTypes.string,
  image: PropTypes.string,
  body: PropTypes.string,
  labelLeft: PropTypes.string,
  labelRight: PropTypes.string,
  children: PropTypes.node,
  actions: PropTypes.node,
  underlayContent: PropTypes.node,
  rowContent: PropTypes.node
};

TableRowComponent.defaultProps = {
  tappable: null,
  swipeable: null,
  selected: null,
  editMode: null,
  header: null,
  image: null,
  size: null,
  title: null,
  subtitle: null,
  body: null,
  icon: null,
  children: null,
  labelLeft: null,
  labelRight: null,
  actions: null,
  underlayContent: null,
  rowContent: null
};

export default TableRowComponent;
