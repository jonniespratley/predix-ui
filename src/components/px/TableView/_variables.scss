@import 'px-spacing-design/_trumps.spacing.scss';

$font-family-default                    : inherit !default;
$font-weight                            : 500 !default;
$table-row-color                        : var(--px-table-row-color, inherit);
$table-row-font-size                    : var(--px-table-row-font-size, $inuit-base-font-size);
$table-row-border-width                 : var(--px-table-row-border-width, 1px);
$table-row-border-style                 : var(--px-table-row-border-style, solid);
$table-row-border-color                 : var(--px-table-row-border-color, black);
$table-row-separator-width              : var(--px-table-row-separator-width, 1px);
$table-row-separator-style              : var(--px-table-row-separator-style, solid);
$table-row-separator-color              : var(--px-table-row-separator-color, black);
$table-row-background-color             : var(--px-table-row-background-color, white);
$table-row-active-background-color      : var(--px-table-row-selected-background-color, black);
$table-row-selected-color               : var(--px-table-row-selected-color, white);
$table-row-selected-background-color    : var(--px-table-row-selected-background-color, black);
$table-row-active-font-color            : var(--px-table-row-selected-color, white);
$table-row-label-font-color             : var(--px-table-row-label-color, black);
$table-row-label-font-size              : var(--px-table-row-label-font-size, $inuit-base-font-size);
$table-row-min-height                   : var(--px-table-row-min-height, calculateRem(44px));
$table-row-line-height                  : var(--px-table-row-line-height, $inuit-base-font-size);
$table-row-header-background-color      : var(--px-table-row-header-background-color, black);
$table-row-header-font-color            : var(--px-table-row-header-color, inherit);
$table-row-title-font-size              : var(--px-table-row-title-font-size, $inuit-base-font-size);
$table-row-title-font-color             : var(--px-table-row-title-color, inherit);
$table-row-subtitle-font-color          : var(--px-table-row-subtitle-color, black);
$table-row-subtitle-font-size           : var(--px-table-row-subtitle-font-size, $inuit-base-font-size);
$table-row-body-font-color              : var(--px-table-row-body-color, black);
$table-row-body-font-size               : var(--px-table-row-body-font-size, $inuit-base-font-size);
$table-row-header-font-size             : var(--px-table-row-header-font-size, $inuit-base-font-size);
$table-row-header-padding               : var(--px-table-row-header-padding, $inuit-base-spacing-unit--tiny);
$table-row-header-text-transform        : var(--px-table-row-header-text-transform, uppercase);
$table-view-responsive-row-width        : calculateRem(230px);
$table-view-responsive-row-gutters      : 0;
$table-view-responsive-row-max          : 4;
$table-view-padding--tiny               : $inuit-base-spacing-unit--tiny;
$table-view-padding--small              : $inuit-base-spacing-unit--small;
$table-view-padding--large              : $inuit-base-spacing-unit--large;
$table-view-padding--huge               : $inuit-base-spacing-unit--huge;
$table-row-button-padding               : $inuit-base-spacing-unit--small;
$table-row-button-margin                : 0;
$table-row-button-color-alert           : var(--px-table-row-button-color-alert, black);
$table-row-button-color-warn            : var(--px-table-row-button-color-warn, black);
$table-row-button-color-primary         : var(--px-table-row-button-color-primary, black);
$table-row-button-color-info            : var(--px-table-row-button-color-info, black);
$table-row-button-color-alert-hover     : var(--px-table-row-button-color-alert-hover, #333);
$table-row-button-color-warn-hover      : var(--px-table-row-button-color-warn-hover, #333);
$table-row-button-color-primary-hover   : var(--px-table-row-button-color-primary-hover, #333);
$table-row-button-color-info-hover      : var(--px-table-row-button-color-info-hover, #333);
$table-row-button-color-alert-active    : var(--px-table-row-button-color-alert-pressed, #555);
$table-row-button-color-warn-active     : var(--px-table-row-button-color-warn-pressed, #555);
$table-row-button-color-primary-active  : var(--px-table-row-button-color-primary-pressed, #555);
$table-row-button-color-info-active     : var(--px-table-row-button-color-info-pressed, #555);
$table-row-button-font-color            : var(--px-table-row-button-font-color, white);
$table-row-button-font-size             : $inuit-base-font-size;
$table-row-actions-transition           : all 0.4s;
$table-row-button-transition            : all 0.4s;
