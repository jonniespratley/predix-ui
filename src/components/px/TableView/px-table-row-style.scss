
@mixin c($el) {
  #{$el} {
    @content;
  }
}

/**
 * px-table-row styles
 */
 .px-table-row {
    @import 'sass/px-table-row-predix';
 }
