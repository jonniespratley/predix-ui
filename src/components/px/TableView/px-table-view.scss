

// Generic
//@import "px-normalize-design/_generic.normalize.scss";
//@import "px-box-sizing-design/_generic.box-sizing.scss";


@mixin c($el) {
  #{$el} {
    @content;
  }
}

/**
 * TableView styles
 */
.px-table-view {
  @import "variables";
  @import "mixins";
  @import "px-mobile-design/sass/_mixins.scss";

  color: var(--px-base-text-color, black);

  // @import 'TableView/sass/TableView-predix.scss';

  .table-row{
    padding: $inuit-base-spacing-unit--tiny;
  }

  ///------------------------------------------------------
  ///.table-view--tiny
  @include modifier ('tiny') {
    @include c ('.table-row') {
      padding: $inuit-base-spacing-unit--tiny !important;
    }
  }

  ///------------------------------------------------------
  ///.table-view--small
  @include modifier ('small') {
    @include c ('.table-row') {
      padding: $inuit-base-spacing-unit--small !important;
    }
  }

  ///------------------------------------------------------
  ///.table-view--large
  @include modifier ('large') {
    @include c ('.table-row') {
      padding: $inuit-base-spacing-unit--large !important;
    }
  }

  ///------------------------------------------------------
  ///.table-view--huge
  @include modifier ('huge') {
    @include c ('.table-row') {
      padding: $inuit-base-spacing-unit--huge !important;
    }
  }


  ///------------------------------------------------------
  ///.table-view--flush
  @include modifier ('flush') {
    @include c ('.table-row') {
      padding: 0 !important;
    }
  }


  // Base
  @include block('table-view') {
    @include clearfix();
    @include pxFonts();
    font-size      : $table-row-font-size;

    list-style-type: none;
    margin         : 0;
    padding        : 0;
    position       : relative;
    display        : flex;
    flex-direction : column;

    ///------------------------------------------------------




    &ul {
      margin: 0 !important;
    }
  }

  // Trumps

}
