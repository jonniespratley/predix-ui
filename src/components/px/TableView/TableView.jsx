import PropTypes from 'prop-types';
import React from 'react';
import styled, { css } from 'styled-components';
import TableRow from './TableRow';

const TableView = styled.ul`
  margin: 0;
  padding: 0;
  list-style: none;

  .px-table-row {
    ${props => props.size === 'flush' && css`
      padding: 0 !important;
    `}
    ${props => props.size === 'tiny' && css`
      padding: 0.33333rem !important;
    `}
    ${props => props.size === 'small' && css`
      padding: 0.66667rem !important;
    `}
    ${props => props.size === 'large' && css`
      padding: 1.33333rem !important;
    `}
    ${props => props.size === 'huge' && css`
      padding: 2rem !important;
    `}
  }
`;

const TableViewComponent = ({
  items,
  tappable,
  size,
  onSelect,
  children
}) => (
  <TableView size={size}>
    {children}
    {items && items.map(item => (
      <TableRow
        tappable={tappable}
        size={size}
        key={item.id}
        {...item}
        onClick={onSelect}
      />
    ))}
  </TableView>
);

TableViewComponent.propTypes = {
  tappable: PropTypes.bool,
  size: PropTypes.string,
  onSelect: PropTypes.func,
  children: PropTypes.node,
  items: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
    title: PropTypes.string,
    icon: PropTypes.string,
    body: PropTypes.string
  }))
};

TableViewComponent.defaultProps = {
  tappable: null,
  size: null,
  onSelect: null,
  children: null,
  items: null
};

export default TableViewComponent;
