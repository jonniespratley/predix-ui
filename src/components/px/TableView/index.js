import React from 'react';
//import { List } from 'react-virtualized';
import classnames from 'classnames';
import styleheet from './px-table-view.scss';
import TableRow from './TableRow';

import styled, {css} from 'styled-components';


/**
 * TableView component
 */

class TableView extends React.Component {
  constructor(props){
    super(props);
    this.displayName = 'TableView';
    this.renderRow = this.renderRow.bind(this);
  }
  _handleSelect(e){
    if(this.props.onSelect){
      this.props.onSelect(e);
    }
  }
  renderRow({ key, index, isScrolling, isVisible, style }) {
    const content = /* isScrolling ? '...' :*/ this.props.items[index];
    return (
      <TableRow 
        tappable={this.props.tappable}
        key={key}
        index={index}
        {...content}
        onClick={this._handleSelect}>
      </TableRow>
    );
  }
  render(){
    const {
      style = {
        outline: 'none'
      },
      items = [],
      selectedItem,
      selected,
      tappable,

      //sizes
      size,
      height = 300,
      width = 320,
      rowHeight = 44,
      children 
    } = this.props;

    const baseClassnames = classnames('px-table-view',
      { 'px-table-view--flush': size === 'flush' },
      { 'px-table-view--tiny': size === 'tiny' },
      { 'px-table-view--small': size === 'small' },
      { 'px-table-view--regular': size === 'regular' },
      { 'px-table-view--large': size === 'large' },
      { 'px-table-view--huge': size === 'huge' }
    );
    const tableViewClassnames = classnames(
      'table-view',
      { 'table-view--flush': size === 'flush' },
      { 'table-view--tiny': size === 'tiny' },
      { 'table-view--small': size === 'small' },
      { 'table-view--regular': size === 'regular' },
      { 'table-view--large': size === 'large' },
      { 'table-view--huge': size === 'huge' }
    );

    return (
      <div className={baseClassnames}>
        {children}
        {items && items.map((item, index) => (
           <TableRow 
            tappable={this.props.tappable}
            size={size}
            key={index}
            index={index}
            {...item}
            onClick={this._handleSelect}>
          </TableRow>
        ))}
      </div>
    );
  }
}
TableView.defaultProps = {
  items: []
};



export default TableView;