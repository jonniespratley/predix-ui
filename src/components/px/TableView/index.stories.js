import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, text, array, object, boolean, number, select } from '@storybook/addon-knobs';

import TableView from './';

function makeRows(count = 5, obj){
	let items = [];
	for(let i = 1; i <= count; i++){
		items.push(Object.assign({
			id: i
		}, obj, {
			title: obj ? `${obj.title} ${i}` : `Item ${i}`
		}));
	}
	return items;
}

const tableSizes = [
	'flush',
	'tiny',
	'small',
	'regular',
	'large',
	'huge'
];

storiesOf('TableView', module)
	.addDecorator(withKnobs)
	.add('default', () => (
		<TableView 
			items={makeRows(5)}
			size={select('size', tableSizes)}
			tappable={boolean('tappable', true)}/>
	))
	.add('with icons', () => (
		<TableView 
			size={select('size', tableSizes)}
			items={makeRows(5, { title: 'Item', icon:'px-nav:home'})}
			tappable={boolean('tappable', true)}
			/>
	))
	.add('with images', () => (
		<TableView 
			size={select('size', tableSizes)}
			items={makeRows(5, { title: 'Item', image:'http://placehold.it/44'})}
			tappable={boolean('tappable', true)}
			/>
	))
	.add('with labels', () => (
		<TableView 
			items={makeRows(5, { 
				title: 'Item', 
				labelRight: 'New'
			})}
			size={select('size', tableSizes)}
			tappable={boolean('tappable', true)}/>
	))
	.add('with descriptions', () => (
		<TableView 
			size={select('size', tableSizes)}
			items={makeRows(5, { 
				title: 'Item', 
				body: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit.',
				labelRight: 'New'
			})}
			tappable={boolean('tappable', true)}/>
	))
	;
