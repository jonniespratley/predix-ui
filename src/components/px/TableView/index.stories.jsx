import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean, select } from '@storybook/addon-knobs';
import { withReadme } from 'storybook-readme';
import README from './README.md';
import TableView from './';
import TableRow from './TableRow';

function makeRows(count = 5, obj) {
  const items = [];
  for (let i = 1; i <= count; i++)/* eslint-disable-line */ {
    items.push(Object.assign({
      id: i
    }, obj, {
      title: obj ? `${obj.title} ${i}` : `Item ${i}`
    }));
  }
  return items;
}

const tableSizes = [
  'flush',
  'tiny',
  'small',
  'regular',
  'large',
  'huge'
];

storiesOf('Components / TableView', module)
  .addDecorator(withKnobs)
  .addDecorator(withReadme(README))
  .addWithJSX('default', () => (
    <TableView
      items={makeRows(5)}
      size={select('size', tableSizes)}
      tappable={boolean('tappable', true)}
    />
  ))
  .addWithJSX('with icons', () => (
    <TableView
      size={select('size', tableSizes)}
      items={makeRows(5, { title: 'Item', icon: 'px-nav:home' })}
      tappable={boolean('tappable', true)}
    />
  ))
  .addWithJSX('with images', () => (
    <TableView
      size={select('size', tableSizes)}
      items={makeRows(5, { title: 'Item', image: 'http://placehold.it/44' })}
      tappable={boolean('tappable', true)}
    />
  ))
  .addWithJSX('with labels', () => (
    <TableView
      items={makeRows(5, {
        title: 'Item',
        labelRight: 'New'
      })}
      size={select('size', tableSizes)}
      tappable={boolean('tappable', true)}
    />
  ))
  .addWithJSX('with descriptions', () => (
    <TableView
      onSelect={action('onSelect')}
      size={select('size', tableSizes)}
      items={makeRows(5, {
        title: 'Item',
        body: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit.',
        labelRight: 'New'
      })}
      tappable={boolean('tappable', true)}
    />
  ))
  .addWithJSX('Table Row', () => (
    <TableRow
      onClick={action('onClick')}
      title={text('title', 'Table Row Title')}
      image={text('image', '')}
      icon={text('icon', '')}
      labelLeft={text('labelLeft', '')}
      labelRight={text('labelRight', '')}
      subtitle={text('subtitle', '')}
      body={text('body', 'Table Row Body')}
      size={select('size', ['flush', 'tiny', 'small', 'regular', 'large', 'huge'])}
      header={boolean('header', false)}
      iconLeft={boolean('iconLeft', false)}
      iconRight={boolean('iconRight', false)}
      swipeable={boolean('swipeable', false)}
      selected={boolean('selected', false)}
      tappable={boolean('tappable', true)}
    />
  ));

