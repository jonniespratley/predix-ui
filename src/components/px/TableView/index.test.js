import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import TableView from './';
import TableRow from './TableRow';

function makeRows(count = 5, obj) {
  const items = [];
  for (let i = 1; i <= count; i++)/* eslint-disable-line */ {
    items.push(Object.assign({
      id: i
    }, obj, {
      title: obj ? `${obj.title} ${i}` : `Item ${i}`
    }));
  }
  return items;
}

describe('TableView', () => {
  test('should render', () =>{
    const tree = renderer.create(<TableView/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should render with "large" rows', () =>{
    const tree = renderer.create(<TableView items={makeRows(5)} size='large'/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('TableRow', () => {
    test('should render', () =>{
      const tree = renderer.create(
        <TableRow 
          title='Row Label' 
          body='This is the body' 
          icon='px-nav:home'/>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
