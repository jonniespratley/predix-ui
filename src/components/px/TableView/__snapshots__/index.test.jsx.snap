// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableView TableRow should render 1`] = `
.c2 {
  color: var(--iron-icon-stroke-color,inherit);
  fill: var(--iron-icon-fill-color,none);
  stroke: var(--iron-icon-stroke-color,currentColor);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  vertical-align: middle;
  width: var(--iron-icon-width);
  height: var(--iron-icon-height);
}

.c2 > svg {
  color: var(--iron-icon-stroke-color,inherit);
  fill: var(--iron-icon-fill-color,none);
  stroke: var(--iron-icon-stroke-color,currentColor);
  position: relative;
  vertical-align: middle;
}

.c2.hidden {
  display: none;
}

.c0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  min-height: var(--px-table-row-min-height,2.93333rem);
  line-height: var(--px-table-row-line-height,15px);
  color: var(--px-table-row-color,inherit);
  font-size: var(--px-table-row-font-size,15px);
  background-color: var(--px-table-row-background-color,white);
  border-bottom-width: var(--px-table-row-separator-width,1px);
  border-bottom-style: var(--px-table-row-separator-style,solid);
  border-bottom-color: var(--px-table-row-separator-color,black);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  position: relative;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-appearance: normal;
  -moz-appearance: normal;
  appearance: normal;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0:after,
.c0:before {
  display: table;
  content: ' ';
}

.c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: normal;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  white-space: nowrap;
  font-size: var(--px-table-row-title-font-size,15px);
  color: var(--px-table-row-title-color,inherit);
}

.c5 {
  font-size: var(--px-table-row-body-font-size,15px);
  -webkit-order: 5;
  -ms-flex-order: 5;
  order: 5;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  white-space: normal;
  line-height: normal;
}

.c1 {
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: var(--px-table-row-media-margin,0);
  margin-right: var(--px-table-row-media-margin,0.66667rem);
}

.c1 img {
  display: block;
  max-width: 80px;
  max-height: 80px;
}

.c3 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  className="px-table-row c0"
  selected={null}
  size={null}
>
  <div
    className="c1"
  >
    <i
      className="px-icon px-nav-home c2"
      color={null}
      style={
        Object {
          "height": 22,
          "width": 22,
        }
      }
    >
      <svg
        dangerouslySetInnerHTML={
          Object {
            "__html": "<g id=\\"px-nav-home\\"><path stroke-linejoin=\\"round\\" d=\\"M3.5 11v10.5h15V11m-18-.5L11 1l10.5 9.5\\"/><path stroke-linejoin=\\"round\\" d=\\"M8.5 15.5h5v6h-5z\\"/></g>",
          }
        }
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "display": "block",
            "height": 22,
            "pointerEvents": "none",
            "width": 22,
          }
        }
        viewBox="0 0 22 22"
      />
    </i>
  </div>
  <div
    className="c3"
  >
    <div
      className="c4"
    >
      Row Label
    </div>
    <div
      className="c5"
    >
      This is the body
    </div>
  </div>
  <div
    className="flex flex--stretch"
    id="underlay"
  />
</div>
`;

exports[`TableView should render 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  list-style: none;
}

<ul
  className="c0"
  size={null}
/>
`;

exports[`TableView should render with "large" rows 1`] = `
.c1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  min-height: var(--px-table-row-min-height,2.93333rem);
  line-height: var(--px-table-row-line-height,15px);
  color: var(--px-table-row-color,inherit);
  font-size: var(--px-table-row-font-size,15px);
  background-color: var(--px-table-row-background-color,white);
  border-bottom-width: var(--px-table-row-separator-width,1px);
  border-bottom-style: var(--px-table-row-separator-style,solid);
  border-bottom-color: var(--px-table-row-separator-color,black);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  position: relative;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-appearance: normal;
  -moz-appearance: normal;
  appearance: normal;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 1.33333rem;
}

.c1:after,
.c1:before {
  display: table;
  content: ' ';
}

.c3 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: normal;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  white-space: nowrap;
  font-size: var(--px-table-row-title-font-size,15px);
  color: var(--px-table-row-title-color,inherit);
}

.c2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  margin: 0;
  padding: 0;
  list-style: none;
}

.c0 .px-table-row {
  padding: 1.33333rem !important;
}

<ul
  className="c0"
  size="large"
>
  <div
    className="px-table-row c1"
    selected={null}
    size="large"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        Item 1
      </div>
    </div>
    <div
      className="flex flex--stretch"
      id="underlay"
    />
  </div>
  <div
    className="px-table-row c1"
    selected={null}
    size="large"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        Item 2
      </div>
    </div>
    <div
      className="flex flex--stretch"
      id="underlay"
    />
  </div>
  <div
    className="px-table-row c1"
    selected={null}
    size="large"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        Item 3
      </div>
    </div>
    <div
      className="flex flex--stretch"
      id="underlay"
    />
  </div>
  <div
    className="px-table-row c1"
    selected={null}
    size="large"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        Item 4
      </div>
    </div>
    <div
      className="flex flex--stretch"
      id="underlay"
    />
  </div>
  <div
    className="px-table-row c1"
    selected={null}
    size="large"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        Item 5
      </div>
    </div>
    <div
      className="flex flex--stretch"
      id="underlay"
    />
  </div>
</ul>
`;
